lib/adiwg/mdtranslator/readers/iso19115_2/modules/module_legal_constraint.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+require 'adiwg/mdtranslator/internal/internal_metadata_obj'
+
+module ADIWG
+   module Mdtranslator
+      module Readers
+         module Iso191152
+            module LegalConstraint
+               # @@legalConstraintXPath = 'gmd:MD_LegalConstraints'
+               @@accessConstraintsXPath = 'gmd:accessConstraints//gmd:MD_RestrictionCode'
+               @@useConstraintsXPath = 'gmd:useConstraints//gmd:MD_RestrictionCode'
+               @@otherConstraintsXPath = 'gmd:otherConstraints//gco:CharacterString'
+               @@type = 'legal'
+               def self.unpack(xLegalConstraint, _hResponseObj)
+                  intMetadataClass = InternalMetadata.new
+                  hConstraint = intMetadataClass.newConstraint
+                  hLegalConstraint = intMetadataClass.newLegalConstraint
+
+                  hConstraint[:type] = @@type
+
+                  # xLegalConstraint = xConstraint.xpath(@@legalConstraintXPath)[0]
+                  return nil if xLegalConstraint.nil?
```

Comment:
we've moved this condition to `Constraint.unpack` so we can delete it from here if you want. no biggie.

Reply:
Will do. Thanks!